src/common/Screen.js

```diff
@@ -151,6 +150,4 @@ class Screen extends PureComponent<Props> {
   }
 }
 
-export default connect(state => ({
-  safeAreaInsets: getSession(state).safeAreaInsets,
-}))(Screen);
+export default withSafeAreaInsets(Screen);
```

Comment:
Hmm -- this seems to lose us type-checking on these props. :-/

Try it by going to a call site of `Screen`, and editing e.g. `title=` to `titl=`. Flow gives no error.

I think one problem here may be that the type of `withSafeAreaInsets` is written with some types that were apparently invented for the TypeScript types for React, like `React.ForwardRefExoticComponent`. I guess those are effectively being treated as `any`, perhaps.

But really the root problem here seems to be within the `react-native-safe-area-context` library, and its TypeScript types. I don't see how this could possibly work to express the actual type of this function:
```js
// node_modules/react-native-safe-area-context/lib/typescript/src/SafeAreaContext.d.ts
export declare function withSafeAreaInsets<T>(WrappedComponent: React.ComponentType<T>): React.ForwardRefExoticComponent<React.PropsWithoutRef<T> & React.RefAttributes<T>>;
```
The inner component should take a prop `insets`, with a certain type... and the outer component should take no such prop. It looks like nothing in that type knows anything about a property name `insets`.

We might be best off with either our own small handwritten libdef, or a type-wrapper. The libdef (or type-wrapper) can have just the type `EdgeInsets` and the types of `withSafeAreaInsets` and `useSafeArea`.

The type of `withSafeAreaInsets` should look like a much simplified version of our `connect` type-wrapper: it should be just like what you get from `connect` if you pass it an argument of type `() => {| insets: EdgeInsets |}`.

Reply:
Ah, thanks for catching that. How about bypassing the problem by using the [`useSafeAreaInsets` hook](https://github.com/th3rdwave/react-native-safe-area-context#usesafeareainsets)?

Reply:
> `Screen` would be quite easy to convert to a Hooks-based component.

Ah, right, but `ComposeBox` wouldn't.

Reply:
I'll see about making a small type-wrapper, so we can use the handy `BoundedDiff` type you wrote. 😄